box-webhook-aws-sample/README.md

```diff
@@ -0,0 +1,104 @@
+
+During the process you have created private key. This key is needed by the Box SDK to authenticate yourself and your App Users. 
+
+###  Setup AWS Environment
+
+As for Amazon Web Services we are going to use the following services:
+
+* **DynamoDB**
+  * non-relational database for storing mapping between user files and notification emails
+  * more information can be found [here](https://aws.amazon.com/dynamodb/)
+* **Simple Notification Service (SNS)**
+  * service used to send notification emails
+  * more information can be found [here](https://aws.amazon.com/sns/)
+* **Lambda**
+  * execution part of the application
+  * manages webhook notification registration in storage as well as triggering email notification
+  * more information can be found [here](https://aws.amazon.com/lambda/)
+* **API Gateway**
+  * web resources for registering webhook notifications and Box Webhooks triggers
+  * more information can be found [here](https://aws.amazon.com/api-gateway/)
+* **Elastic Beanstalk**
+  * easy-to-use deployment of our example web application on Amazon
+  * more information can be found [here](https://aws.amazon.com/elasticbeanstalk/)
+* **CloudFormation**
+  * gives an easy way to create and manage a collection of related AWS resources
+  * service used to prepare our Sample AWS Environment
+
+#### 1. Connect your Box Application with the Sample Application
+
+In order for the sample application to work with your Box Application you have created in previous steps, you will need to fill the following values.
+Rename `gradle.properties.template` file in sample root to `gradle.properties` and replace placeholders with values for your application and enterprise account.
+
+```
+{boxEnterpriseId}         enterprise ID of your Box account
+{boxClientId}             client ID of the Box Application
+{boxClientSecret}         client secret of the Box Application
+{boxPublicKeyId}          ID of your public key in the Box Application
+{boxPrivateKeyPassword}   password for your private key
+{boxPrivateKeyFile}       absolute path of your private key file
+```
+You are now ready to build your packages.
+
+#### 2. Preparing Application Packages for AWS
+
+In order to get packages we need to configure AWS environment use Gradle build task in root directory of the sample application:
+
+For Mac/Linux:
+```sh
+./gradlew clean build
+```
+
+For Windows:
+```cmd
+gradlew.bat clean build
+```
+
+The build process will build WAR file for our Web Application to be deployed on Elastic Beanstalk, ZIP file for Lambda function and also CloudFormation configuration file.
+Files can be found in `build/distributions/aws-s3` directory.
+
+#### 3. Uploading packages to S3
+
+ We are now ready to setup AWS environment. We will do so by using CloudFormation. To make our packages accessible to CloudFormation, we will upload them to S3 bucket.
+
+ 1. Go to [S3 Console](https://console.aws.amazon.com/s3) and Create New Bucket
```

Comment:
I tried running this as an IAM user I created with Power user and it failed with:
because it was unable to create an IAM user.   So you may want mention that the AWS user needs administror privledges